test/e2e/specs/scaleupdown.go

```diff
@@ -0,0 +1,216 @@
+package specs
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"github.com/ghodss/yaml"
+	apiappsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	internal "github.com/openshift/openshift-azure/pkg/api"
+	"github.com/openshift/openshift-azure/pkg/api/2018-09-30-preview/api"
+	"github.com/openshift/openshift-azure/pkg/util/randomstring"
+	"github.com/openshift/openshift-azure/pkg/util/ready"
+	"github.com/openshift/openshift-azure/test/clients/azure"
+	"github.com/openshift/openshift-azure/test/clients/openshift"
+)
+
+var _ = Describe("Scale Up/Down E2E tests [ScaleUpDown] [Real] [LongRunning] [EveryPR]", func() {
+	var (
+		azurecli *azure.Client
+		occli    *openshift.Client
+		manifest = flag.String("manifest", "../../test/manifests/normal/scaleup.yaml", "Path to the scale up manifest to send as a partial update to the RP")
+		update   = flag.String("update", "../../test/manifests/normal/scaledown.yaml", "Path to the scale down manifest to send as a partial update to the RP")
+	)
+
+	const (
+		sampleDeployment = "hello-openshift"
+	)
+
+	BeforeEach(func() {
+		var err error
+		azurecli, err = azure.NewClientFromEnvironment()
+		Expect(err).NotTo(HaveOccurred())
+		occli, err = openshift.NewCustomerAdminClient()
+		Expect(err).NotTo(HaveOccurred())
+	})
+
+	It("should be possible to maintain a healthy cluster after scaling it out and in", func() {
+		ctx := context.Background()
+		ctx = context.WithValue(ctx, internal.ContextKeyClientID, os.Getenv("AZURE_CLIENT_ID"))
+		ctx = context.WithValue(ctx, internal.ContextKeyClientSecret, os.Getenv("AZURE_CLIENT_SECRET"))
+		ctx = context.WithValue(ctx, internal.ContextKeyTenantID, os.Getenv("AZURE_TENANT_ID"))
+
+		By("Fetching the scale up manifest")
+		external, err := parseExternalConfig(*manifest)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(external).NotTo(BeNil())
+
+		By("Calling update on the rp with the scale up manifest")
+		updated, err := azurecli.OpenShiftManagedClusters.CreateOrUpdate(ctx, os.Getenv("RESOURCEGROUP"), os.Getenv("RESOURCEGROUP"), *external)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(updated).NotTo(BeNil())
+
+		By("Creating a namespace as cluster admin")
+		namespace, err := randomstring.RandomString("abcdefghijklmnopqrstuvwxyz0123456789", 5)
+		Expect(err).ToNot(HaveOccurred())
+		namespace = "e2e-test-" + namespace
+		err = occli.CreateProject(namespace)
```

Comment:
Wouldn't it be clearer to move project creation in BeforeEach and project deletion in AfterEach?